custom/icds_reports/utils/aggregation.py

```diff
@@ -1535,6 +1536,381 @@ def indexes(self, aggregation_level):
+            ('tt_1', 'ut.due_list_date_tt_1'),
+            ('tt_2', 'ut.due_list_date_tt_2'),
+            ('immediate_breastfeeding', 'agg_bp.immediate_breastfeeding'),
+            ('anemia', 'agg_bp.anemia'),
+            ('eating_extra', 'agg_bp.eating_extra'),
+            ('resting', 'agg_bp.resting'),
+            ('anc_weight', 'agg_bp.anc_weight'),
+            ('anc_blood_pressure', 'agg_bp.anc_blood_pressure'),
+            ('bp_sys', 'agg_bp.bp_sys'),
+            ('bp_dia', 'agg_bp.bp_dia'),
+            ('anc_hemoglobin', 'agg_bp.anc_hemoglobin'),
+            ('bleeding', 'agg_bp.bleeding'),
+            ('swelling', 'agg_bp.swelling'),
+            ('blurred_vision', 'agg_bp.blurred_vision'),
+            ('convulsions', 'agg_bp.convulsions'),
+            ('rupture', 'agg_bp.rupture'),
+            ('bp_date', 'agg_bp.latest_time_end_processed::DATE'),
+            ('is_ebf', 'agg_pnc.is_ebf'),
+            ('breastfed_at_birth', 'agg_delivery.breastfed_at_birth'),
+            ('person_name', 'case_list.person_name'),
+            ('edd', 'case_list.edd'),
+            ('delivery_nature', 'case_list.delivery_nature'),
+            ('mobile_number', 'case_list.mobile_number'),
+            ('preg_order', 'case_list.preg_order'),
+            ('num_pnc_visits', 'case_list.num_pnc_visits'),
+        )
+        return """
+        INSERT INTO "{tablename}" (
+            {columns}
+        ) (SELECT
+            {calculations}
+            FROM "{ucr_ccs_record_monthly_table}" ucr
+            LEFT OUTER JOIN "{agg_thr_table}" agg_thr ON ucr.doc_id = agg_thr.case_id AND ucr.month = agg_thr.month and ucr.valid_in_month = 1
+            LEFT OUTER JOIN "{agg_bp_table}" agg_bp ON ucr.doc_id = agg_bp.case_id AND ucr.month = agg_bp.month and ucr.valid_in_month = 1
+            LEFT OUTER JOIN "{agg_pnc_table}" agg_pnc ON ucr.doc_id = agg_pnc.case_id AND ucr.month = agg_pnc.month and ucr.valid_in_month = 1
+            LEFT OUTER JOIN "{agg_delivery_table}" agg_delivery ON ucr.doc_id = agg_delivery.case_id AND ucr.month = agg_delivery.month and ucr.valid_in_month = 1
+            LEFT OUTER JOIN "{ccs_record_case_ucr}" case_list ON ucr.doc_id = case_list.doc_id
+            LEFT OUTER JOIN "{pregnant_tasks_case_ucr}" ut ON ucr.doc_id = ut.ccs_record_case_id
+            WHERE ucr.month = %(start_date)s
+            ORDER BY ucr.awc_id, ucr.case_id
+        )
+        """.format(
+            tablename=self.tablename,
+            columns=", ".join([col[0] for col in columns]),
+            calculations=", ".join([col[1] for col in columns]),
+            ucr_ccs_record_monthly_table=self.ccs_record_monthly_ucr_tablename,
+            agg_thr_table=AGG_CCS_RECORD_THR_TABLE,
+            ccs_record_case_ucr=self.ccs_record_case_ucr_tablename,
+            agg_pnc_table=AGG_CCS_RECORD_PNC_TABLE,
+            agg_bp_table=AGG_CCS_RECORD_BP_TABLE,
+            agg_delivery_table=AGG_CCS_RECORD_DELIVERY_TABLE,
```

Comment:
Since all of these tables haven't been fully rebuilt,  does that mean that all old columns use their old definitions? (e.g. there aren't any changes to old column calculations in this PR)

Reply:
there are no changes to old column calculations. only new columns